packages/opentelemetry-metrics/src/export/types.ts

```diff
@@ -102,16 +113,68 @@ export interface MetricExporter {
 /**
  * Base interface for aggregators. Aggregators are responsible for holding
  * aggregated values and taking a snapshot of these values upon export.
+ *
+ * Use {@link Aggregator} instead of this BaseAggregator.
  */
-export interface Aggregator {
+interface BaseAggregator {
+  /** The kind of the aggregator. */
+  kind: AggregatorKind;
+
   /** Updates the current with the new value. */
   update(value: number): void;
+}
+
+/** SumAggregatorType aggregate values into a {@link Sum} point type. */
+export interface SumAggregatorType extends BaseAggregator {
```

Comment:
All those interfaces `SumAggregatorType`, `LastValueAggregatorType` etc. don't have to exist.
It is enough to have just existing `Aggregator` (no need for `BaseAggregator` too) to do everything you need and to have correct type when calling `toPoint`. What's the point in having so many different interfaces if they all looks the the same and in final class when `implements` certain interface you have to specify the type again. Having field in interface 
```
kind: AggregatorKind.SUM
``` 
doesn't make sense the interface is meant to define types for each field so it should be
```
kind: AggragatorKind;
```

And here is the piece of code you are using when implementing a final class
```
export class MinMaxLastSumCountAggregator
  implements DistributionAggregatorType {
  public kind: AggregatorKind.DISTRIBUTION = AggregatorKind.DISTRIBUTION;
  toPoint(): Point<Distribution> {
```
No need for those interfaces, this is just wrong approach.
I have mentioned the same here
https://github.com/open-telemetry/opentelemetry-js/pull/1310#discussion_r456137083

Reply:
> It is enough to have just existing Aggregator (no need for BaseAggregator too) to do everything you need and to have correct type when calling toPoint

If i'm not mistaken you'll have to cast the return of `toPoint` to get the exact type that this aggregator return. This approach allows to have the correct `toPoint` type just by checking the the kind of aggregator.

Reply:
@vmarchaud the compiler and any ide with suggestion would already know that see my whole example from here
https://github.com/open-telemetry/opentelemetry-js/pull/1310#discussion_r456137083

Reply:
[Well it doesn't](https://www.typescriptlang.org/play/#code/FAUwdgrgtgBAggcwQJxAghgFwPbICoCeADiDAN7AwwDKAqgLIA0lMAMnNXgPoBqcrtAKLMqAEQCSnAEriAQrTziA8gDkRMABKS8SgOJS4TYAF9gwTMVIaAlgGccKdLAC8Me8mtgE5yzWgxXSCgAIxBkM09MMIAzdABjUgAFbEjCEgAePBgQAA8osAATWzdMDy8YAB8YINDkAD5yFgA3dAAbCBAALhg8EwiwKORYhJhZdFsQRBQ0LFxGqgBrTwLuqdQMHHxLMyocZMiACgBKbrIYFvauv1gqm3tsR1hTU2BImPjSamg1mc3svPARVG40mSHWs2Q8xgS0KqzBv1waRAADo6PQdjA9ikBsduvsBkj0jUwnU+q8Bu8RncHMgnD8NnNcvkgWMJvSIVCYSt4PCGVsSMitJw9AZ0SwsYcTjB8ZhCe5PAhSS8LCQedM+QFNHYaXTeRyql8oOzNmY4q1xsVqQ9aUa9ZtxFAiK0YNZHa0QFBwJhLdrrbr1RyKItlnCA5skYLtCLDJrjYjLJHhfpDBiJTijlCqKhMBBkGByOc2h1ugByKL2EswUxUZ6m7BgewwdBITVgEAAdy190eceQDqdxybxV7r2iMAOzYQyK5AWcrl7EaFOmT9AzQZgAHoNzB660CCUyt4qJPkWnMMdkRcOmSt+dbGZ+oNhqRewAmTkhtXg8PbcXYGW4gWV5XIalRdjqTxkmaFrgX6tphrg76uk6HpesOdqIR+sJfgi-Ioku0b0LGGF4Ymy6iqm-7YueUoynKpQKg065ZiAOZ5kBRZXGWIAVlWLC1nE9aNpO76BB2sE9iR-atIO4w4Xyr6juOInTsss7ziRi5Riua4sLe8rlHMxLhMeSCnlRkqXpxJhAA) in the playground, or atleast what you suggested.

Reply:
it is working fine 
[playground](https://www.typescriptlang.org/play/?ssl=1&ssc=1&pln=126&pc=1#code/JYWwDg9gTgLgBAbzgCSgFVAUzgXzgMyghDgHIABCMTAOxkwBtMRMYoBPAegEMxhSA3ACghMdtTgBlAK4kAvHBqyARpijCxEgDLcAzjABq3BtOwKlIVepHA6a-NwDG2ACLB9UYMukxgEGohCcHAgtgBciipqwsEg3AAeERZWMXAMejBJUdbBjhDSdFmW0UFwurJFKUI4NnZQDs4o7jAQAOZQ3CQIpd6OANasuhHdwcHK+TQAJtyemEORxVAA2gC6qbkTMPPJaqupOKnlIJUlGwWZC1U1QrSycACCre2YrdwtUADStpOBwZIAqgBZOAKUhSIGkAA0pUBAEkAHIAfUB9wAGoitPdJGhEQDAYiAMIAeX+8LQILIcKRKPRmOxuKBhJJZKhpRcsOxACVYQAhf5oWFE+EU0jsrm8-mC+Gs4LIDloIkAcU592BoLl2KVKsBrOummwj2er3eaHEcwpMhIAB84Dp9EYTNgbW4PF4fH4AjbkM02h0QMIhLZ6PUnAanlAXm9oL84H1vhFDRHjdAvlMA8FpGBpvQABQAN2MphOUAAlBE8xBgJN03AWgAFSt0HNluANoMAHkTkZNZt0AD5hNcg-ZQ63GzB22g+zGC46ImhUr4WPpOmAIqgMCxByJHOldLopLIu8moHBQGAmCw6Afj1HTyNY98Kbf3qnJgA6PE1zPZzD5wuYMWLYVlWiDXME9bjs2ERtnQ7aWtOD7BBGMDSFAARIaMs5FnAACM0KjKMS5zDAq4REsAAMkJwBRKwEaMBwiMENTXLuegHoCtiAgkdowJaBKbC+0bnpetBbA84bdtGD5xlMz6SSeb7vlSyJohiWI4niTKkguTFwD+bx-thgGXGowGVj8CDgbWECwTA0Fjh2LpsG6vj+IhpTIawaEYZ5WEAcMfmEaENARLCND4LYwBiPRhGxAkEQALThZFNDRewsVxXuFxUUFoxHBEuVxaMeTnIVmXMRVtZYCu4DkVRNF0X5jHMdUO7+PoIRcTxGT8YJCl3hSNCYAA7nAnE0Nx8S8X15xCVAzbCCFU0zbIAlzQN7zvgZua4SWwh5DQnWQEGuEUstPX6LNdDze+kFBotQiHboEBMO+DBtDmJ10Lh77Ge9GT7XAnCcIoEBwGARDKJeB40ODw2YD8LRwI4GTtUd8BHPNQ2jYeIDzY9WObdA21ZoZOZ7QdHXwN9MAAEwUkTRp3ndtlQftT0da9mDvZ9tN039AEc0IIO1hwISYCLoPPdzvOtF944C-92VAyD4lqEQUAiKLPKckSHwAKLCvNB7I3majAPg7BnvgtYHiN0B9AeqWS1LBDAPELAS3AyjW9wkyTLYrRkPgEAQGCyPzW+T17je4bcCeADq0UABY8kQAw0F8PwicwYlx8z7wxrJPygqH4ffmTubGUB5YWWBen3U2LZ2fBsgeYRKE+TGhE13hVXEbVa5wJR1G0ZlLW4G1gZ1A02AAGJhzGygzMW25u5FnvYF7vtwP7gc0MH5c2RJhfQKa1C6DH7Gnx0Sep+nECZ3ZZ7gKJ1631J96lCX8ln583xPxAkrr+f8c5TKljrqBKyjc2YPRbuOdsi8IAd1GF3dCPd-LgMwoRFeUAIi4TyjgAeNVSJ1RHg1cezV9htSAA)
and on stackblitz
[stackblitz](https://codesandbox.io/s/restless-http-5cff6?file=/index.ts)

Reply:
There is a misunderstanding on the issue here; when exported to exporter you only get the `Aggregator` and that [doesn't work with your code](https://www.typescriptlang.org/play/?ssl=83&ssc=1&pln=127&pc=1#code/JYWwDg9gTgLgBAbzgCSgFVAUzgXzgMyghDgHIABCMTAOxkwBtMRMYoBPAegEMxhSA3ACghMdtTgBlAK4kAvHBqyARpijCxEgDLcAzjABq3BtOwKlIVepHA6a-NwDG2ACLB9UYMukxgEGohCcHAgtgBciipqwsEg3AAeERZWMXAMejBJUdbBjhDSdFmW0UFwurJFKUI4NnZQDs4o7jAQAOZQ3CQIpd6OANasuhHdwcHK+TQAJtyemEORxVAA2gC6qbkTMPPJaqupOKnlIJUlGwWZC1U1QrSycACCre2YrdwtUADStpOBwZIAqgBZOAKUhSIGkAA0pUBAEkAHIAfUB9wAGoitPdJGhEQDAYiAMIAeX+8LQILIcKRKPRmOxuKBhJJZKhpRcsOxACVYQAhf5oWFE+EU0jsrm8-mC+Gs4LIDloIkAcU592BoLl2KVKsBrOummwj2er3eaHEcwpMhIAB84Dp9EYTNgbW4PF4fH4AjbkM02h0QMIhLZ6PUnAanlAXm9oL84H1vhFDRHjdAvlMA8FpGBpvQABQAN2MphOUAAlBE8xBgJN03AWgAFSt0HNluANoMAHkTkZNZt0AD5hNcg-ZQ63GzB22g+zGC46ImhUr4WPpOmAIqgMCxByJHOldLopLIu8moHBQGAmCw6Afj1HTyNY98Kbf3qnJgA6PE1zPZzD5wuYMWLYVlWiDXME9bjs2ERtnQ7aWtOD7BBGMDSFAARIaMs5FnAACM0KjKMS5zDAq4REsAAMkJwBRKwEaMBwiMENTXLuegHoCtiAgkdowJaBKbC+0bnpetBbA84bdtGD5xlMz6SSeb7vlSyJohiWI4niTKkguTFwD+bx-thgGXGowGVj8CDgbWECwTA0Fjh2LpsG6vj+IhpTIawaEYZ5WEAcMfmEaENARLCND4LYwBiPRhGxAkEQALThZFNDRewsVxXuFxUUFoxHBEuVxaMeTnIVmXMRVtZYCu4DkVRNF0X5jHMdUO7+PoIRcTxGT8YJCl3hSNCYAA7nAnE0Nx8S8X15xCVAzbCCFU0zbIAlzQN7zvgZua4SWwh5DQnXcE8FLLT1+izXQ81wHoElGneQiHZ1kBBrhFIna076QUGi1PR1EBMO+DBtDmr10Lh77GcDGT7XAnCcIoEBwGARDKJeB40Mjw2YD8LRwI4GRCEAA)

Reply:
what doesn't work ?